src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                        port
+                    } else {
+                        DEFAULT_PORT.to_string().parse().unwrap()
+                    };
+            }
+        } else {
+            if !self.configuration.manager_bind_host.is_empty() {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-bind-host".to_string(),
+                    value: self.configuration.manager_bind_host.clone(),
+                    detail: "The --manager-bind-host option can not be set together with the --worker flag.".to_string(),
+                });
+            }
+
+            if self.configuration.manager_bind_port != 0 {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-bind-port".to_string(),
+                    value: self.configuration.manager_bind_port.to_string(),
+                    detail: "The --manager-bind-port option can not be set together with the --worker flag.".to_string(),
+                });
+            }
+        }
+
+        // Configure manager_host and manager_port.
+        if self.attack_mode == GooseMode::Worker {
+            // Use default if run-time option not set.
+            if self.configuration.manager_host.is_empty() {
+                self.configuration.manager_host =
+                    if let Some(host) = self.defaults.manager_host.clone() {
+                        host
+                    } else {
+                        "127.0.0.1".to_string()
+                    }
+            }
+
+            // Use default if run-time option not set.
+            if self.configuration.manager_port == 0 {
+                self.configuration.manager_port = if let Some(port) = self.defaults.manager_port {
+                    port
+                } else {
+                    DEFAULT_PORT.to_string().parse().unwrap()
+                };
+            }
+        } else {
+            if !self.configuration.manager_host.is_empty() {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-host".to_string(),
+                    value: self.configuration.manager_host.clone(),
+                    detail:
+                        "The --manager-host option must be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+
+            if self.configuration.manager_port != 0 {
+                return Err(GooseError::InvalidOption {
+                    option: "--manager-port".to_string(),
+                    value: self.configuration.manager_port.to_string(),
+                    detail:
+                        "The --manager-port option must be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    // Configure how many Goose Users to hatch.
+    fn set_users(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd In this case we are writing information in `GooseAttack` based on `configuration` and `defaults`. This could also just also overload `configuration`, as we track how many users _actually_ start in the metrics structure.

Reply:
Actually this can not use `configuration` as Users is optional. The final value lives in `GooseAttack` so we can detect if someone incorrectly tries to configure `--users` on a Worker.